activerecord/lib/active_record/relation/batches.rb

```diff
@@ -45,14 +46,15 @@ module Batches
     #
     # NOTE: You can't set the limit either, that's used to control
     # the batch sizes.
-    def find_each(start: nil, batch_size: 1000)
+    def find_each(start: nil, end_at: nil, batch_size: 1000)
       if block_given?
-        find_in_batches(start: start, batch_size: batch_size) do |records|
+        find_in_batches(start: start, end_at: end_at, batch_size: batch_size) do |records|
           records.each { |record| yield record }
         end
       else
-        enum_for(:find_each, start: start, batch_size: batch_size) do
-          start ? where(table[primary_key].gteq(start)).size : size
+        enum_for(:find_each, start: start, batch_size: batch_size, end_at: end_at) do
```

Comment:
Let's insert `end_at` before `batch_size` as above for consistency (it's nice that start and end is together) :smile: